examples/calling-apis/spa-with-backend-api/create-agent-chat-app/README.md

```diff
@@ -0,0 +1,325 @@
+# LangGraph Memory Agent with Auth0 AI and Token Vault
+
+This example demonstrates a **Memory Agent** chat application built with [LangGraph](https://langchain-ai.github.io/langgraph/) that integrates [Auth0 AI](https://auth0.ai) for authentication and third-party API access through Token Vault. The agent maintains conversation history, remembers user preferences, and can access external APIs like Google Calendar on behalf of authenticated users.
+
+This project was generated using the public [`langchain-ai/create-agent-chat-app`](https://github.com/langchain-ai/create-agent-chat-app) generator and enhanced with Auth0 authentication and Token Vault integration for secure third-party API access.
+
+<div align="center">
+  <img src="public/create-agent-chat-app.gif" alt="View the Memory Agent demo" />
+</div>
+
+## What is the Memory Agent?
+
+The Memory Agent is an intelligent conversational assistant that:
+
+- **Remembers conversations**: Stores and retrieves user interactions and preferences across sessions
+- **Learns about users**: Builds a knowledge base about user preferences, habits, and context
+- **Accesses external APIs**: Integrates with third-party services like Google Calendar using secure token exchange
+- **Maintains context**: Uses stored memories to provide personalized responses and recommendations
+
+The agent can help with tasks like:
+- Scheduling and calendar management
+- Remembering personal preferences and past conversations
+- Providing contextual recommendations based on user history
+- Coordinating with external services while maintaining security
+
+## Features
+
+This template leverages a modern stack for building a full-stack agent application with authentication:
+
+- **LangGraph Memory Agent**: Persistent memory management with conversation history
+- **Auth0 Authentication**: Secure user authentication with Auth0 SPA integration
+- **Token Vault Integration**: Secure third-party API access without exposing sensitive credentials
+- **Google Calendar Integration**: Access user calendar data through OAuth2 with Auth0 connections
+- **React Frontend**: Modern UI built with React, TypeScript, and Tailwind CSS
+- **Monorepo Structure**: Organized with Turbo for build orchestration
+- **Custom Auth Handler**: Implements Auth0 JWT validation for LangGraph API
+- **Works Everywhere**: Compatible with both self-hosted and LangGraph Platform deployments
+
+## Authentication & Token Exchange Flow
+
+This application implements Auth0's Token Vault for secure third-party API access:
+
+![Access Token Exchange Flow](../react-hono-ai-sdk/public/access-token-exchange-token-vault.png)
```

Comment:
this diagram needs updating. Will upload a new one shortly...

Reply:
updated:
https://github.com/auth0-lab/auth0-ai-js/blob/b2989bd785872ff72f3a67432c02566ca6a1d111/examples/calling-apis/spa-with-backend-api/create-agent-chat-app/README.md